sbc_email/models/import_letters_history.py

```diff
@@ -379,28 +380,114 @@ def _get_smb_connection(self):
+
+        lang_id = None
+        if language_name:
+            # Retrieve language name and find corresponding id
+            model_lang = self.env['res.lang.compassion'].search(
+                [('name', '=', language_name)], limit=1)
+
+            lang_id = model_lang.id
+            logger.info(
+                "language name found {}".format(
+                    model_lang.display_name))
+
+        if template_name:
+            # Retrieve template name and find corresponding id
+            model_template = self.env['correspondence.template'].search(
+                [('name', '=', template_name)], limit=1)
+
+            template = None
+
+            template_id = model_template.id
+            logger.info("template id found {}".format(template_id))
+
+            template = model_template.browse(template_id)
+
+            if template:
+                logger.info("template found: {}".format(template.name))
+            else:
+                template = self.env.ref('sbc_compassion.default_template')
+
+        if not original_text:
+            original_text = ""
 
-        if import_web_letter:
-            logger.info('Import letter with config Web Letter existence')
+        # save_letter pdf
+        if pdf_letter:
+            file_pdf = BytesIO(base64.b64decode(pdf_letter))
+
+            filename = 'WEB_' + sponsor_ref + '_' + \
+                child_code + '_' + str(time.time())[:10] + '.pdf'
+
+            logger.info("Try to copy file {} !".format(filename))
+
+            # Copy file in attachment in the done letter folder
+            share_nas = self.env.ref('sbc_email.share_on_nas').value
+            done_letter_path = self.env.ref(
+                'sbc_email.scan_letter_done').value + filename
+
+            smb_conn = self._get_smb_connection()
+            if smb_conn and smb_conn.connect(
+                    SmbConfig.smb_ip, SmbConfig.smb_port):
+                smb_conn.storeFile(share_nas, done_letter_path, file_pdf)
+                smb_conn.close()
         else:
-            logger.info('NO IMPORT letter with config Web Letter')
-            import_config = self.env['import.letter.config'].search(
-                [('name', '=', 'Web Letter')], limit=1)
-            import_web_letter = self.create({
-                'config_id': import_config.id,
-                'state': 'open'})
-
-        # TODO Create import letter
+            return False
+
+        # analyze attachment to check template and create image preview
+        line_vals, document_vals = func.analyze_attachment(
+            self.env, base64.b64decode(pdf_letter), filename,
+            template)
+
+        for i in xrange(0, len(line_vals)):
+            line_vals[i].update({
+                'import_id': self.id,
+                'partner_id': sponsor_id,
+                'child_id': child_id,
+                'letter_language_id': lang_id,
+                'original_text': original_text,
+                'status': 'ok'
```

Comment:
Why do you force the status of the line ? If lang is not correctly given, or template is default, the status shouldn't be ok. I think it is better to remove this.